workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/HandleCreatePlayerRequestSystem.cs

```diff
@@ -12,83 +12,86 @@ namespace Improbable.Gdk.PlayerLifecycle
     [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
     public class HandleCreatePlayerRequestSystem : ComponentSystem
     {
-        private struct CreatePlayerData
-        {
-            public readonly int Length;
-            [ReadOnly] public ComponentDataArray<PlayerCreator.CommandRequests.CreatePlayer> CreatePlayerRequests;
-            [ReadOnly] public ComponentDataArray<WorldCommands.CreateEntity.CommandSender> CreateEntitySender;
-        }
+        private CommandSystem commandSystem;
 
-        [Inject] private CreatePlayerData createPlayerData;
-
-        private struct EntityCreationResponseData
+        protected override void OnCreateManager()
         {
-            public readonly int Length;
-            [ReadOnly] public ComponentDataArray<WorldCommands.CreateEntity.CommandResponses> CreateEntityResponses;
-            [ReadOnly] public ComponentDataArray<PlayerCreator.CommandResponders.CreatePlayer> CreatePlayerResponders;
+            base.OnCreateManager();
+            commandSystem = World.GetExistingManager<CommandSystem>();
         }
 
-        [Inject] private EntityCreationResponseData entityCreationResponseData;
-
         private class PlayerCreationRequestContext
         {
             public PlayerCreator.CreatePlayer.ReceivedRequest createPlayerRequest;
         }
 
         protected override void OnUpdate()
         {
-            for (var i = 0; i < createPlayerData.Length; i++)
+            HandlePlayerCreateRequests();
+            HandlePlayerCreateEntityResponses();
+        }
+
+        private void HandlePlayerCreateRequests()
+        {
+            if (PlayerLifecycleConfig.CreatePlayerEntityTemplate == null)
             {
-                var requests = createPlayerData.CreatePlayerRequests[i].Requests;
-                var createEntitySender = createPlayerData.CreateEntitySender[i];
+                throw new InvalidOperationException(Errors.PlayerEntityTemplateNotFound);
+            }
 
-                foreach (var request in requests)
-                {
-                    if (PlayerLifecycleConfig.CreatePlayerEntityTemplate == null)
+            var requests = commandSystem.GetRequests<PlayerCreator.CreatePlayer.ReceivedRequest>();
+            foreach (var request in requests)
```

Comment:
This is about to be broken by moving to spans. Would be better to make this a regular for like, as for component updates.

Reply:
@samiwh Should this be done before merging? Or fixed when the change happens?

Reply:
Rebased on spans API